api/app/src/main/kotlin/packit/controllers/PacketController.kt

```diff
@@ -43,23 +38,29 @@ class PacketController(private val packetService: PacketService)
 
     @GetMapping("/{id}/file")
     @PreAuthorize("@authz.canReadPacket(#root, #id)")
-    fun findFile(
+    fun streamFile(
         @PathVariable id: String,
         @RequestParam hash: String,
         @RequestParam inline: Boolean = false,
         @RequestParam filename: String,
-    ): ResponseEntity<ByteArrayResource>
-    {
+        response: HttpServletResponse
+    ) {
         val packet = packetService.getMetadataBy(id)
         if (packet.files.none { it.hash == hash }) {
             throw PackitException("doesNotExist", HttpStatus.NOT_FOUND)
         }
 
-        val response = packetService.getFileByHash(hash, inline, filename)
-        return ResponseEntity
-            .ok()
-            .headers(response.second)
-            .body(response.first)
+        val disposition = if (inline) "inline" else "attachment"
+        val mediaType = getMediaType(filename).orElse(MediaType.APPLICATION_OCTET_STREAM)
+        val headers = HttpHeaders().apply {
+            contentType = MediaType.valueOf(mediaType.toString())
```

Comment:
```suggestion
            contentType = mediaType
```

or am I missing something?